contracts/verifiers/UniversalVerifier.sol

```diff
@@ -317,4 +320,31 @@ contract UniversalVerifier is OwnableUpgradeable, IUniversalVerifier {
     ) public view returns (uint256) {
         return _getMainStorage().proofs[user][requestId].storageFields[key];
     }
+
+    /// @notice Gets the list of reuest IDs and verifies the proofs are linked
+    /// @param requestIds the list of reuest IDs
+    /// @return Whether the proofs are linked
+    function verifyLinkedProofs(uint64[] calldata requestIds) public view returns (bool) {
+        Proof storage firstProof = _getMainStorage().proofs[_msgSender()][requestIds[0]];
+        uint256 linkIdToCompare = firstProof.storageFields[LINKED_PROOF_KEY];
+
+        if (linkIdToCompare == 0) {
+            revert("linkID cannot be 0.");
+        }
+
+        for (uint256 i = 1; i < requestIds.length; i++) {
+            Proof storage proof = _getMainStorage().proofs[_msgSender()][requestIds[i]];
+            uint256 linkID = proof.storageFields[LINKED_PROOF_KEY];
+
+            if (linkID == 0) {
+                revert("linkID cannot be 0.");
+            }
+
+            if (linkIdToCompare != linkID) {
+                revert("linkID verification failed");
```

Comment:
we need to find a way to specify specific requests have specific link id but they are not linked